tests/unit/test_charm.py

```diff
@@ -4,64 +4,317 @@
 # Learn more about testing at: https://juju.is/docs/sdk/testing
 
 import unittest
-from unittest.mock import Mock
+from unittest.mock import Mock, patch
 
 from ops.model import ActiveStatus
 from ops.testing import Harness
 
-from charm import OperatorTemplateCharm
+from charm import PgBouncerK8sCharm
+
+INI_PATH = "/etc/pgbouncer/pgbouncer.ini"
+USERLIST_PATH = "/etc/pgbouncer/userlist.txt"
 
 
 class TestCharm(unittest.TestCase):
     def setUp(self):
-        self.harness = Harness(OperatorTemplateCharm)
+        self._pgbouncer_container = "pgbouncer"
+
+        self.harness = Harness(PgBouncerK8sCharm)
         self.addCleanup(self.harness.cleanup)
         self.harness.begin()
 
-    def test_config_changed(self):
-        self.assertEqual(list(self.harness.charm._stored.things), [])
-        self.harness.update_config({"thing": "foo"})
-        self.assertEqual(list(self.harness.charm._stored.things), ["foo"])
+    def test_on_install(self):
+        self.harness.charm._on_install("mock_event")
```

Comment:
You can also change to `self.harness.charm.on.install.emit()` to assert that `on_install` hook is correctly observed on charm's `__init__`. It's not required, just an enhancement.